ignite/distributed/comp_models/base.py

```diff
@@ -206,17 +217,21 @@ def _collective_op(
             return self._decode_str(tensor)
         return tensor
 
-    def all_reduce(self, tensor: Union[torch.Tensor, float], op: str = "sum") -> Union[torch.Tensor, float]:
+    def all_reduce(
+        self, tensor: Union[torch.Tensor, float], op: str = "sum", group: Optional[List[List[int]]] = None
```

Comment:
Same here, why group type hint is `Optional[List[List[int]]]`, I think we need support 2 options:
```
Optional[Union[Any, List[int]]]
```
What do you think ?

Reply:
Seems more concise and better !